### PR TITLE
App Config Fixing Live Tests

### PR DESCRIPTION
**Pull request by mrm9084**
# Description

Fixing Live App Config Tests

### CODE REVIEW

File: sdk/appconfiguration/test-resources.json [hunk 1]
```diff
@@ -125,10 +126,67 @@
...
-              "[resourceId('Microsoft.KeyVault/vaults', parameters('baseName'))]"
+                "[resourceId('Microsoft.KeyVault/vaults', parameters('baseName'))]"
             ],
             "properties": {
-              "value": "Very secret value"
+                "value": "Very secret value"
+            }
+        },
+        {
+            "type": "Microsoft.AppConfiguration/configurationStores/keyValues",
```
Comment (YalinLi0312) [inline sdk/appconfiguration/test-resources.json:136]: Is it possible to set these keyvalues programmatically in test? We usually setup basic resources in this file, not preparing data for tests.

Comment (mrm9084) [reply sdk/appconfiguration/test-resources.json:136]: I am currently doing it in both. But I think this is the correct way to do it. How else do you validate that it actually works? See https://github.com/Azure/azure-sdk-for-python/issues/31326, you have tests for feature flags, but they don't actually work. You are just validating you are consist within the SDK not with the service.

File: sdk/appconfiguration/azure-appconfiguration-provider/tests/async_preparers.py [hunk 2]
```diff
@@ -12,6 +12,9 @@ async def wrapper(*args, **kwargs):
         appconfiguration_connection_string = kwargs.pop("appconfiguration_connection_string")
         kwargs["appconfiguration_connection_string"] = appconfiguration_connection_string
 
+        appconfiguration_keyvault_secret_url = kwargs.pop("appconfiguration_keyvault_secret_url")
+        kwargs["appconfiguration_keyvault_secret_url"] = appconfiguration_keyvault_secret_url
```
Comment (xiangyan99) [inline sdk/appconfiguration/azure-appconfiguration-provider/tests/async_preparers.py:16]: What's this for?

**Review by xiangyan99 — COMMENTED** [review]
We want to move recordings to another repo. Don't we? @YalinLi0312

File: sdk/appconfiguration/azure-appconfiguration-provider/tests/async_preparers.py (hunk 2 above)
Comment (mrm9084) [reply sdk/appconfiguration/azure-appconfiguration-provider/tests/async_preparers.py:16]: I actually don't know what that section does. I was just copying what was in the other preparers.

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**